app/institutions/dashboard/-components/object-list/component-test.ts

```diff
@@ -0,0 +1,169 @@
+            },
+        };
+        this.setProperties({
+            columns,
+            institution,
+            defaultQueryOptions,
+            objectType: 'thingies',
+        });
+    });
+
+    test('the table headers are correct', async function(assert) {
+        await render(hbs`
+<Institutions::Dashboard::-Components::ObjectList
+    @columns={{this.columns}}
+    @institution={{this.institution}}
+    @defaultQueryOptions={{this.defaultQueryOptions}}
+    @objectType={{this.objectType}}
+/>
+        `);
+
+        // Elements from InstitutionDashboarWrapper are present
+        assert.dom('[data-test-link-to-reports-archive]').exists('Link to download prior reports exists');
+        assert.dom('[data-test-page-tab="summary"]').exists('Summary tab exists');
+
+        // Elements in the top bar are present
+        assert.dom('[data-test-object-count]').containsText('10 total thingies', 'Object count is correct');
+        assert.dom('[data-test-toggle-filter-button]').exists('Filter button exists');
+        assert.dom('[data-test-customize-columns-button]').exists('Columns button exists');
+
+        assert.dom('[data-test-object-list-table]').exists('Object list exists');
+
+        // The table headers are correct
+        assert.dom('[data-test-column-header]').exists({ count: 4 }, 'There are 4 columns');
+        assert.dom('[data-test-column-header="Title"]').containsText('Title');
+        assert.dom('[data-test-column-header="Title"] [data-test-sort="title"]').exists('Title is sortable');
+        assert.dom('[data-test-column-header="Description"]').containsText('Description');
+        assert.dom('[data-test-column-header="Description"] [data-test-sort="description"]')
+            .doesNotExist('Description is not sortable');
+
+        // The table data is not blatantly incorrect
+        assert.dom('[data-test-object-table-body-row]').exists({ count: 10 }, 'There are 10 rows');
+    });
+
+    test('the table supports filtering', async function(assert) {
+        await render(hbs`
+<Institutions::Dashboard::-Components::ObjectList
+    @columns={{this.columns}}
+    @institution={{this.institution}}
+    @defaultQueryOptions={{this.defaultQueryOptions}}
+    @objectType={{this.objectType}}
+/>
+        `);
+
+        await click('[data-test-toggle-filter-button]');
+
+        assert.dom('[data-test-filter-facet-toggle]').exists({ count: 3 }, '3 filters available');
+
+        // Open the filter facet and load the values and select the first filter value
+        await click('[data-test-filter-facet-toggle]');
+        await click('[data-test-filter-facet-value] button');
+
+        assert.dom('[data-test-active-filter]').exists({ count: 1 }, '1 filter active');
+        assert.dom('[data-test-remove-active-filter]').exists('Remove filter button exists');
+
+        await click('[data-test-remove-active-filter]');
+        assert.dom('[data-test-active-filter]').doesNotExist('Filter removed');
+    });
```

Comment:
Should this test include checking that the filtered out rows are no longer visible?

Reply:
Unfortunately, our current mirage view (mirage/views/search.ts) is fairly simple and doesn't really do anything with the filter query-params. It just serves up randomized data each time. I could try to make it a little more sophisticated so it can actually filter out certain results, but I worry that it may be a bit of an undertaking for relatively little gain.

Reply:
Gotcha. Probably not worth it then.